### PR TITLE
store the denom in map for performance

### PR DESCRIPTION
**Pull request by juniuszhou**

### CODE REVIEW

File: x/tokenregistry/keeper/keeper.go [hunk 1]
```diff
@@ -85,75 +85,67 @@ func (k keeper) CheckDenomPermissions(ctx sdk.Context, denom string, requiredPer
 	return true
 }
 
+func (k keeper) GetDenomPrefix(ctx sdk.Context, denom string) []byte {
```
Comment (smartyalgo) [inline x/tokenregistry/keeper/keeper.go:88]: Does this need to be visible outside of the package?

Comment (smartyalgo) [reply x/tokenregistry/keeper/keeper.go:88]: If not, I think we can make it lowercase

File: x/tokenregistry/keeper/keeper.go [hunk 2]
```diff
@@ -85,75 +85,67 @@ func (k keeper) CheckDenomPermissions(ctx sdk.Context, denom string, requiredPer
...
-	}
 
-	return types.RegistryEntry{
-		IsWhitelisted: false,
-		Denom:         denom,
-	}
+	var entry types.RegistryEntry
+	key := k.GetDenomPrefix(ctx, denom)
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(key)
```
Comment (smartyalgo) [inline x/tokenregistry/keeper/keeper.go:97]: bz is hard to understand, can you update this var name to be more descriptive?

File: x/tokenregistry/keeper/keeper.go [hunk 3]
```diff
@@ -85,75 +85,67 @@ func (k keeper) CheckDenomPermissions(ctx sdk.Context, denom string, requiredPer
 	return true
 }
 
+func (k keeper) GetDenomPrefix(ctx sdk.Context, denom string) []byte {
+	return append(types.TokenDenomPrefix, []byte(denom)...)
+}
+
 func (k keeper) GetDenom(ctx sdk.Context, denom string) types.RegistryEntry {
```
Comment (smartyalgo) [inline x/tokenregistry/keeper/keeper.go:92]: This func, if denom is not found, returns a RegistryEntry with Denom = denom.

Are we preserving it in this revision? I suggest adding a unit test to encapsulate this behaviour

**Review by smartyalgo — CHANGES_REQUESTED** [review]

File: x/tokenregistry/keeper/keeper.go [hunk 4]
```diff
@@ -85,75 +85,67 @@ func (k keeper) CheckDenomPermissions(ctx sdk.Context, denom string, requiredPer
...
-		if wl.Entries[i] != nil && strings.EqualFold(wl.Entries[i].Denom, denom) {
-			return *wl.Entries[i]
-		}
-	}
 
-	return types.RegistryEntry{
-		IsWhitelisted: false,
-		Denom:         denom,
-	}
+	var entry types.RegistryEntry
```
Comment (banshee) [inline x/tokenregistry/keeper/keeper.go:94]: GetDenom needs to return a valid struct if there's no key stored for the denom

**Comment by banshee** [comment]
It turns out that develop has changed this so much we're going to need to rework it anyway.  Let's take this PR for peggy2 and do the work necessary when we merge in develop to future/peggy2.

**Review by banshee — APPROVED** [review]

File: x/tokenregistry/keeper/keeper.go (hunk 1 above)
Comment (juniuszhou) [reply x/tokenregistry/keeper/keeper.go:88]: should be lowercase, not used outside.

File: x/tokenregistry/keeper/keeper.go (hunk 4 above)
Comment (juniuszhou) [reply x/tokenregistry/keeper/keeper.go:94]: accepted.

**Pull request merged**